src/main/java/org/opentripplanner/util/ThrowableUtils.java

```diff
@@ -16,4 +18,13 @@ public static String detailedString (Throwable throwable) {
         return sw.toString();
     }
 
+    public static void unwrapAndRethrowCompletionException(CompletionException e) {
+        if (e.getCause() instanceof RoutingValidationException) {
+            throw (RoutingValidationException) e.getCause();
+        } else if (e.getCause() instanceof RuntimeException) {
+            throw (RuntimeException) e.getCause();
+        }
+        throw e;
+    }
```

Comment:
This break module dependencies. The new method have dependencies to domain, this creates circular dependencies.

Reply:
Would you prefer to duplicate the code?

Reply:
Just a reflection: This is of cause DRY, but it introduce other problems. How do we know this exist? Where do we look for it? The natural place to put this would be in `concurrent` in "fremwork support", but we have no such thing in OTP.

Reply:
Put it in the `RoutingValidationException` that is what has the strongest coupling to it.